src/include/btree.h

```diff
@@ -130,6 +126,30 @@ struct __wt_btree {
 	uint64_t rec_max_txn;		/* Maximum txn seen (clean trees) */
 	uint64_t write_gen;		/* Write generation */
 
+	enum {
+		WT_CKPT_OFF, WT_CKPT_PREPARE, WT_CKPT_RUNNING
+	} checkpointing;		/* Checkpoint in progress */
+
+	/*
+	 * We flush pages from the tree (in order to make checkpoint faster),
+	 * without a high-level lock.  To avoid multiple threads flushing at
+	 * the same time, lock the tree.
+	 */
+	WT_SPINLOCK	flush_lock;	/* Lock to flush the tree's pages */
+
+	bool modified;			/* If the tree ever modified */
+	bool bulk_load_ok;		/* Bulk-load is a possibility */
+
+	/*
+	 * The tree's cache and eviction information persist past after the
+	 * handle is closed (clean cache pages may remain after the tree is
+	 * closed). Be careful clearing the WT_BTREE structure.
+	 */
+#define	WT_BTREE_CLEAR_SIZE(b)						\
+	(WT_PTRDIFF(&(b)->root, b))
```

Comment:
> The bigger issue is why do we need to keep the fields below on a closed tree (particularly since they will be cleared on next open)?  Ideally we could free `dhandle->handle` in that case -- can't we just assert that eviction isn't active in a closed tree?

There's some code in `__wt_conn_btree_sync_and_close` that I don't entirely understand, but it's possible to get through there without discarding the contents of the tree, and with `WT_DHANDLE_DEAD` set on the handle, which I think implies we've closed the tree but eviction will still be active in the tree.

We could maybe fix this by not discarding the `WT_BTREE` structure until we're discarding the contents of the cache, but the obvious changes to do that immediately break with:
```
lsm-worker: wt-000003.lsm: file-remove: file has open handles.
```

We could also move the eviction information from the `WT_BTREE` structure into the `WT_DATA_HANDLE` structure (or allocate an eviction structure on demand if that makes the data-handle too large).